src/net/db.rs

```diff
@@ -1275,12 +1276,16 @@ impl PeerDB {
 
         if always_include_allowed {
             // always include allowed neighbors, freshness be damned
-            let allow_qry = "SELECT * FROM frontier WHERE network_id = ?1 AND denied < ?2 AND (allowed < 0 OR ?3 < allowed) AND (peer_version & 0x000000ff) >= ?4".to_string();
+            // the peer_version check mirrors the check in `has_acceptable_epoch`:
+            //    (my_epoch <= peer_epoch) OR (curr_epoch <= peer_epoch)
+            let allow_qry = "SELECT * FROM frontier WHERE network_id = ?1 AND denied < ?2 AND \
+            (allowed < 0 OR ?3 < allowed) AND (?4 <= (peer_version & 0x000000ff) OR ?5 <= (peer_version & 0x000000ff))".to_string();
```

Comment:
I'm not sure this is what we want here.  First, it seems logically incorrect -- I think you'd want `(?4 & 0x000000ff) <= (peer_version & 0x000000ff)`; otherwise this clause is always false because `peer_version` includes the network's major and minor version numbers in higher bits.  Second, it's always true that `(curr_network_epoch as u32) <= (peer_version & 0x000000ff)` because the node's `peer_version` reflects the highest epoch supported by the software.  The `has_acceptable_epoch` code performs both checks because the first check is a short-circuit that saves the code from performing the slightly-more-expensive second check.

Reply:
Regarding your second comment: peer_version is a neighbor's version, so I don't see why (curr_network_epoch <= peer_epoch) would necessarily always be true. 

An example:
my_epoch = 2.05 
curr_epoch = 2.0 
peer_epoch = 2.0

Because of the second check (curr_epoch <= peer_epoch), the peer would be found to have an acceptable epoch prior to the start of Epoch 2.05, but after the node upgraded.

Reply:
Ah, yes, you're right about my second comment.

The first part needs to be addressed though.